ogmios/__init__.py

```diff
@@ -101,20 +106,16 @@ def get_recipients(self, name):
     def get_subject(self):
         return self.render_string(self.data['subject'])
 
-    def get_attachments(self):
-        error = EmailTemplateError("Attachments should be a list of either "
-                                   "strings or a tuple "
-                                   "(name, filename, mimetype)")
-
-        for attachment in self.data.get('attachments', []):
-            if isinstance(attachment, six.string_types):
-                yield self.render_string(attachment)
-            elif isinstance(attachment, collections.Iterable):
-                if len(attachment) != 3:
-                    raise error
-                yield tuple(self.render_string(s) for s in attachment)
+    def validate_attachments(self):
+        for attachment in self.attachments:
+            if set(attachment.keys()) == set(['path', 'name', 'type']):
```

Comment:
Why not support `{'path': ..., 'name': ...}` and a string. Passing a dict for just one  field seems complicated. And in the future we could support file-like objects.